### ISSUE TITLE
Reproduction Problems

### ISSUE THREAD

**Issue by KaXuan1135** [WuYichen-97/SD-Lora-CL#4]
Hi, I have some questions about reproducing the results in paper.

1. In the paper, you state that the hyperparameter being used for ImageNet-R dataset is Adam as optimizer, 8e-3 learning rate and 30 epochs, but in your provided log files, some of them follows these settings while other of them use 1e-2 learning rate and 20 epochs, which one setting is the one that can reproduce the result on ImageNet-R in the paper?  

2. In the code 'sdlora.py', the optimizer is fixed to SGD regardless of the one stated in config file, is that an implementation error or that's what it is originally.

3. In the paper, you stated that the pretrained model being used is ViT-B/16 pretrained on ImageNet-21K and fine-tuned on ImageNet-1K, and I also observed that this pretrained model is being used in the provided log file. But I did not see any implementation of using this pretrained model in current provided code, and by manually the loading this pretrained model, I observed the reduction on performance of SD-LoRA, what did I do wrong here?

I really appreciate the effort and time taken to provide the paper, code, and logs—they are immensely helpful for understanding and reproducing your work. It would mean a lot if you could help clarify the above points. Thank you in advance for your guidance. Looking forward to your insights!

**Comment by linlany** [comment]
I’ve also been working on reproducing the results and would like to add a related point.

I successfully reproduced the ImageNet-R results using the provided config files. However, I'm currently observing a performance drop of about 1.65% on CIFAR-100 compared to the reported results in the paper. It would be great if the authors could look into that discrepancy.

Also, as far as I can tell, the current code uses the timm library to load the pretrained ViT-B/16 model. This model is pretrained on ImageNet-21K and fine-tuned on ImageNet-1K, which aligns with what's stated in the paper.

Looking forward to further clarification from the authors. Thanks again for sharing the code and logs!

**Comment by KaXuan1135** [comment]
> I’ve also been working on reproducing the results and would like to add a related point.
> 
> I successfully reproduced the ImageNet-R results using the provided config files. However, I'm currently observing a performance drop of about 1.65% on CIFAR-100 compared to the reported results in the paper. It would be great if the authors could look into that discrepancy.
> 
> Also, as far as I can tell, the current code uses the timm library to load the pretrained ViT-B/16 model. This model is pretrained on ImageNet-21K and fine-tuned on ImageNet-1K, which aligns with what's stated in the paper.
> 
> Looking forward to further clarification from the authors. Thanks again for sharing the code and logs!

Hi, can you provide your log file for reference, and what settings did you use to reproduce the result. (Optimizer, Learning Rate and epochs). Did you fixed the problem of the 'SGD optimizer will be used regardless of the one stated in config file'? Thanks in advance.

I also want to apologize for my earlier assumption. I initially thought that ViT-B/16 pretrained on ImageNet-21K and fine-tuned on ImageNet-1K specifically referred to 'timm/vit_base_patch16_224.augreg_in21k', given its presence in the provided log files. However, if the currently loaded model 'vit_base_patch16_224' serves the same purpose, that should be fine.

Thank you in advance for your time and clarification!

**Comment by linlany** [comment]
> > I’ve also been working on reproducing the results and would like to add a related point.
> > I successfully reproduced the ImageNet-R results using the provided config files. However, I'm currently observing a performance drop of about 1.65% on CIFAR-100 compared to the reported results in the paper. It would be great if the authors could look into that discrepancy.
> > Also, as far as I can tell, the current code uses the timm library to load the pretrained ViT-B/16 model. This model is pretrained on ImageNet-21K and fine-tuned on ImageNet-1K, which aligns with what's stated in the paper.
> > Looking forward to further clarification from the authors. Thanks again for sharing the code and logs!
> 
> Hi, can you provide your log file for reference, and what settings did you use to reproduce the result. (Optimizer, Learning Rate and epochs). Did you fixed the problem of the 'SGD optimizer will be used regardless of the one stated in config file'? Thanks in advance.
> 
> I also want to apologize for my earlier assumption. I initially thought that ViT-B/16 pretrained on ImageNet-21K and fine-tuned on ImageNet-1K specifically referred to 'timm/vit_base_patch16_224.augreg2_in21k_ft_in1k', given its presence in the provided log files. However, if the currently loaded model 'vit_base_patch16_224' serves the same purpose, that should be fine.
> 
> Thank you in advance for your time and clarification!

I didn't change the code and config. The log file is confusing because it is multiple and the experiment is continuously appended. The configuration is as follows:
{
    "prefix": "test",
    "dataset": "imagenetr",
    "memory_size": 0,
    "memory_per_class": 0,
    "fixed_memory": false,
    "shuffle": true,
    "init_cls": 20,
    "increment": 20,
    "model_name": "sdlora",
    "backbone_type": "vit_base_patch16_224",
    "device": ["0"],
    "optimizer": "adam",
    "scheduler": "constant",
    "filepath": "./ImageNetR3/",
    "seed": [1995],
    "init_epoch": 20,
    "init_lr": 1e-2,
    "init_milestones": [10000],
    "init_lr_decay": 0,
    "init_weight_decay": 0.0005,
    "epochs": 20,
    "lrate": 1e-2,
    "milestones": [10000],
    "lrate_decay": 0,
    "batch_size": 128,
    "weight_decay": 2e-4
}

**Comment by KaXuan1135** [comment]
Thx, I have successfully reproduced results that are close to those outlined in the paper, though some settings deviated from what was specified in the paper. I will now proceed to close this issue.

**Comment by linlany** [comment]
> Thx, I have successfully reproduced results that are close to those outlined in the paper, though some settings deviated from what was specified in the paper. I will now proceed to close this issue.

Can the results be reproduced on the cifar dataset and with what configs

**Comment by WuYichen-97** [comment]
Hi Linlany, sorry for missing your question. I've uploaded the C100.log file for your reference.

**Comment by bnufw** [comment]
@KaXuan1135 @linlany Hello, I would like to ask how to organize imagenet-r and imagenet-a into train and val, and what is the general ratio for allocation?
